xmtp_mls/src/identity.rs

```diff
@@ -104,7 +103,8 @@ impl IdentityStrategy {
                 legacy_signed_private_key,
             ) => {
                 if let Some(stored_identity) = stored_identity {
-                    if inbox_id != stored_identity.inbox_id {
+                    let checksum_inbox_id = generate_inbox_id_from_checksum(&address, &nonce)?;
```

Comment:
If the inbox ID was created with the checksum address, and not the lowercased one, it shouldn't be able to create a valid Create Inbox identity action

Reply:
We check that the inbox ID matches the signer as part of verifying that Identity Action, so it should blow up there. 

So even if your client "works", everyone else would reject you